### PR TITLE
digitalocean_kubernetes_cluster version changes

### LINKED ISSUE #332: Updating Kubernetes cluster version should not cause replacement
Hi there,

Changing the version of a Kubernetes cluster should not recreate the resource.

An upgrade API is available: https://developers.digitalocean.com/documentation/v2/#upgrade-a-kubernetes-cluster

### Affected Resource(s)
Please list the resources as a list, for example:
- digitalocean_kubernetes_cluster

### Expected Behavior
Kubernetes cluster upgrades without being replaced.

### Actual Behavior
The Kubernetes cluster is replaced.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Change the `version` attribute of a `digitalocean_kubernetes_cluster`
2. `terraform apply`

### PR DESCRIPTION
**Pull request by aqche**
Changing the `version` on a `digitalocean_kubernetes_cluster` will perform an upgrade instead of forcing a new resource.

Fixes #332 

```
aqche$ make testacc TESTARGS='-run=TestAccDigitalOceanKubernetesCluster_UpgradeVersion'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run=TestAccDigitalOceanKubernetesCluster_UpgradeVersion -timeout 120m
?       github.com/terraform-providers/terraform-provider-digitalocean  [no test files]
=== RUN   TestAccDigitalOceanKubernetesCluster_UpgradeVersion
=== PAUSE TestAccDigitalOceanKubernetesCluster_UpgradeVersion
=== CONT  TestAccDigitalOceanKubernetesCluster_UpgradeVersion
--- PASS: TestAccDigitalOceanKubernetesCluster_UpgradeVersion (376.99s)
PASS
ok      github.com/terraform-providers/terraform-provider-digitalocean/digitalocean
```

### CODE REVIEW

**Comment by botzill** [comment]
Thx for this PR, really waiting for it!

**Comment by eddiezane** [comment]
@aqche thanks for this PR!

I’m wondering if we should use something like the custom diff function as opposed to going this route. There are some side effects by dropping the force new and hoping the upgrade request does what’s wanted.

@snormore @andrewsomething  thoughts?

**Comment by aqche** [comment]
@eddiezane thanks for the comment! I added a custom diff check so force destroy will occur when version is decreased instead of just erroring.

File: digitalocean/resource_digitalocean_kubernetes_cluster.go [hunk 1]
```diff
@@ -41,10 +43,13 @@ func resourceDigitalOceanKubernetesCluster() *schema.Resource {
 			},
 
 			"version": {
-				Type:         schema.TypeString,
-				Required:     true,
-				ForceNew:     true,
-				ValidateFunc: validation.NoZeroValues,
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
```
[the remark this answers is not in the document]
Comment (aqche) [reply digitalocean/resource_digitalocean_kubernetes_cluster.go]: makes sense, reverted back to `validation.NoZeroValues`

File: digitalocean/resource_digitalocean_kubernetes_cluster_test.go [hunk 2]
```diff
@@ -422,7 +423,35 @@ func TestAccDigitalOceanKubernetesCluster_KubernetesProviderInteroperability(t *
...
+				Config: testAccDigitalOceanKubernetesConfigBasic(rName, testClusterVersion15),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckDigitalOceanKubernetesClusterExists("digitalocean_kubernetes_cluster.foobar", &k8s),
+					resource.TestCheckResourceAttr("digitalocean_kubernetes_cluster.foobar", "version", testClusterVersion15),
+				),
+			},
+			{
+				Config: testAccDigitalOceanKubernetesConfigBasic(rName, testClusterVersion16),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckDigitalOceanKubernetesClusterExists("digitalocean_kubernetes_cluster.foobar", &k8s),
```
[the remark this answers is not in the document]
Comment (aqche) [reply digitalocean/resource_digitalocean_kubernetes_cluster_test.go]: good call, added your suggested change

**Comment by aqche** [comment]
@andrewsomething Thanks for the review! Updated the code to address your comments. 

```
aqche$ make testacc TESTARGS='-run=TestAccDigitalOceanKubernetesCluster_UpgradeVersion'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run=TestAccDigitalOceanKubernetesCluster_UpgradeVersion -timeout 120m
?       github.com/terraform-providers/terraform-provider-digitalocean  [no test files]
=== RUN   TestAccDigitalOceanKubernetesCluster_UpgradeVersion
=== PAUSE TestAccDigitalOceanKubernetesCluster_UpgradeVersion
=== CONT  TestAccDigitalOceanKubernetesCluster_UpgradeVersion
--- PASS: TestAccDigitalOceanKubernetesCluster_UpgradeVersion (296.28s)
PASS
ok      github.com/terraform-providers/terraform-provider-digitalocean/digitalocean     296.314s
```

**Comment by ddollar** [comment]
Thanks for implementing this!

**Comment by eddiezane** [comment]
:tada: @aqche thanks again for the PR!

Would you please shoot me an email when you get a chance? My GitHub username at digitalocean dot com.

**Comment by aqche** [comment]
@andrewsomething Thanks for making the finishing touches on the PR! 🙏

@ddollar Happy to help!

@eddiezane Will do!

**Pull request merged**